bin/build.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+# This script buildd BlazingMQ and all of its dependencies.
+#
+# Required prerequisites:
+# - git
+# - Perl
+# - Python3
+# - Clang
+# - CMake
+# - pkg-config
+# - Homebrew
+
+set -e
+set -u
+set -x
+[ -z $BASH ] || shopt -s expand_aliases
+
+
+# :: Set some initial constants :::::::::::::::::::::::::::::::::::::::::::::::
+DIR_ROOT="${DIR_ROOT:-`pwd`}"
+DIR_THIRDPARTY="${DIR_ROOT}/thirdparty"
+
+
+# :: Download source-code of our dependencies :::::::::::::::::::::::::::::::::
+if [ ! -d "${DIR_ROOT}/blazingmq" ]; then
+    git clone https://github.com/bloomberg/blazingmq.git
+fi
+if [ ! -d "${DIR_THIRDPARTY}/bde-tools" ]; then
+    git clone https://github.com/bloomberg/bde-tools "${DIR_THIRDPARTY}/bde-tools"
+fi
+if [ ! -d "${DIR_THIRDPARTY}/bde" ]; then
+    git clone https://github.com/bloomberg/bde.git "${DIR_THIRDPARTY}/bde"
+fi
+if [ ! -d "${DIR_THIRDPARTY}/ntf-core" ]; then
+    git clone https://github.com/bloomberg/ntf-core.git "${DIR_THIRDPARTY}/ntf-core"
+fi
+
+DIR_BUILD="${DIR_ROOT}/blazingmq/cmake.bld/`uname`"
+mkdir -p ${DIR_BUILD}
```

Comment:
`mkdir -p "${DIR_BUILD}"` in case `${DIR_ROOT}` has spaces (yuck); _mutatis mutandis_ for a few cases below

Reply:
d